### ISSUE TITLE
timeout when attach to a busy process (100% cpu)

### ISSUE THREAD

**Issue by yinrong** [kmaork/madbg#21]
madbg fails to attach, and the running process become DEAD after a while!
```
$ madbg attach 127108
Traceback (most recent call last):
  File "/home/work/hdd1/yinrong/.conda/envs/torch/bin/madbg", line 8, in <module>
    sys.exit(cli())
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/madbg/__main__.py", line 29, in attach
    attach_to_process(pid, port)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/madbg/api.py", line 31, in attach_to_process
    connect_to_debugger(ip, port, timeout=connect_timeout)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/madbg/client.py", line 64, in connect_to_debugger
    with connect_to_server(ip, port, timeout) as socket:
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/home/work/hdd1/yinrong/.conda/envs/torch/lib/python3.9/site-packages/madbg/client.py", line 56, in connect_to_server
    raise TimeoutError()
TimeoutError
(torch) exit-status: 1
```

rerun 'madbg attach', the running process exits:
```
Waiting for connection from debugger console on 127.0.0.1:3513
Waiting for connection from debugger console on 127.0.0.1:3513
OSError: [Errno 98] Address already in use

```

I use madbg since I want to know what the process is busy doing, however madbg never works in such situation..

**Comment by yinrong** [comment]
I changed the timeout parameter in source code and now it works.

**Comment by kmaork** [comment]
@yinrong , I've released madbg 1.2.0 in which I increased the default timeout and added timeout control to the CLI
